framework/include/actions/ActionFactory.h

```diff
@@ -91,11 +65,9 @@ class ActionFactory
 
   InputParameters getValidParams(const std::string & name);
 
-  class BuildInfo
+  struct BuildInfo
   {
-  public:
-    buildActionPtr _build_pointer;
-    paramsActionPtr _params_pointer;
+    std::shared_ptr<RegistryEntryBase> _obj_pointer;
```

Comment:
That'd be a pain in the butt. The objects are co-owned between the Registry and the Action Factory. We could do naked pointers, since the teardown is broken in clang and we need to rely on the OS to free the memory when the application exits. IMO this is fine, as there are no other resources to be freed and we need the registry and factory until the program quits.

Reply:
Sounds good. thanks for the explanation; just making sure